javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -100,38 +100,39 @@ public static String annotationSimpleName(AnnotationMirror annotation) {
      */
     public static boolean areSame(
             /*@Nullable*/ AnnotationMirror a1, /*@Nullable*/ AnnotationMirror a2) {
-        if (a1 != null && a2 != null) {
-            if (a1 == a2) {
-                return true;
-            }
-            if (annotationName(a1) != annotationName(a2)) {
-                return false;
-            }
-
-            Map<? extends ExecutableElement, ? extends AnnotationValue> elval1 =
-                    getElementValuesWithDefaults(a1);
-            Map<? extends ExecutableElement, ? extends AnnotationValue> elval2 =
-                    getElementValuesWithDefaults(a2);
+        if (a1 == null || a2 == null) {
+            return a1 == a2;
+        }
```

Comment:
One important case is missing: both `a1` and `a2` being non-null and `==`.
This changed logic now requires converting the same object to a string twice, which will slow things down.